### PR TITLE
Support typescript

### PR DESCRIPTION
**Pull request by jerncomania28**
- Added Typescript support .
- Converted EMI Calculator components to use typescript.
- migrated EMI stylings to tailwindcss.
-  implemented a sample of a reusable way to make API Request leveraging react-query.

### CODE REVIEW

**Comment by mohitkhatri06** [comment]
as for now in UI getting these errors
image

**Comment by bharathkumarchowdary** [comment]
> as for now in UI getting these errors
> image

@jerncomania28 , Do you too see the same error?
I remember you said there is a way to partially introduce typescript to project
If that is valid, I think you can introduce it to the parts you can solve for now.. no rush!

**Comment by jerncomania28** [comment]
I Don't  happen to encounter any of the errors and from what I'm seeing , it's the way you're going about The conversion , i modified the tsconfig to recognize react and accommodate javascript without type-checking , this is a typing errors , observe how I convert the files and use typing refer to index.tsx in the EMI_Calculator folder under pages and the FormInput component for reference .

Also , because the charts is leveraging a library you might have to install it's accompanying types usually something like @types/name-of-library

**Comment by bharathkumarchowdary** [comment]
> I Don't happen to encounter any of the errors and from what I'm seeing , it's the way you're going about The conversion , i modified the tsconfig to recognize react and accommodate javascript without type-checking , this is a typing errors , observe how I convert the files and use typing refer to index.tsx in the EMI_Calculator folder under pages and the FormInput component for reference .
> 
> Also , because the charts is leveraging a library you might have to install it's accompanying types usually something like @types/name-of-library

@mohitkhatri06 , do we have any typing package for the charts as @jerncomania28 is suggesting?
@jerncomania28 , 

- I too see some errors in local env after checking out this branch. Uploaded to dropbox [here](https://www.dropbox.com/s/d2l8jt894ukac74/support_typescript_errors.pdf?dl=0)
- The build for this branch is failing, can you see what is the reason?

**Comment by mohitkhatri06** [comment]
@bharathkumarchowdary the file in which you mentioned the error at dropbox is module not found, we need to run 'npm install' as there are some package added into the project, and that needs to be added into your node_modules local folder.

For charts there is a package https://www.npmjs.com/package/@types/recharts as mentioned by @jerncomania28 , but the errors which are there is for all the components Support, Expenses etc.

**Comment by mohitkhatri06** [comment]
@bharathkumarchowdary when we click on the notification bell all notifications are visible but to close this, should we have concept where we can close that notification by clicking anywhere on the screen? as for now we need to click again on that bell icon.

**Comment by bharathkumarchowdary** [comment]
> @bharathkumarchowdary when we click on the notification bell all notifications are visible but to close this, should we have concept where we can close that notification by clicking anywhere on the screen? as for now we need to click again on that bell icon.

Yes @mohitkhatri06 , @jerncomania28 ,
Clicking anywhere else should close that modal
I think this behaviour is also expected of the feedback button. Maybe @jerncomania28 can inherit that behaviour from it

File: src/pages/EmiCalculator/index.tsx [hunk 1]
```diff
@@ -70,20 +73,26 @@ const EmiCalculator = () => {
...
-          password: process.env.REACT_APP_PASSWORD as string,
-        },
-      });
-    } catch (error) {
-      console.error("Error:", error);
-      alert("Error occurred during API call.");
-    } finally {
-      setIsLoading(false);
-    }
+    // setIsLoading(true);
```
Comment (mohitkhatri06) [inline src/pages/EmiCalculator/index.tsx:76]: @jerncomania28  as you commented these line it will not break other parts of it right?

**Comment by mohitkhatri06** [comment]
getting this error.
image

**Comment by bharathkumarchowdary** [comment]
but still the tests failed

**Review by bharathkumarchowdary — DISMISSED** [review]

**Comment by mohitkhatri06** [comment]
getting following error in the sign up page also

https://github.com/GVR-Secure-Solutions/lekkalu-frontend/assets/78607953/769b9581-08fe-4e45-9177-3563684d5d7c

**Review by ivangarcilazo — CHANGES_REQUESTED** [review]
Hi Jeremiah, I get these errors: In expenses the fetchTags error and the income statement error are not loading the data.
Jeremiah
Jeremiah 2

**Comment by bharathkumarchowdary** [comment]
@ivangarcilazo , @mohitkhatri06 , do you have further comments on this PR?

**Comment by mohitkhatri06** [comment]
when trying to add new expense, the following error comes up. 
image

and the expense table is not  visible.

**Pull request closed without merging**